go/src/pluginmgr/pluginmgr_common/pluginmgr_helper.go

```diff
@@ -0,0 +1,505 @@
+	getPlugin(plgname string) (plugins_common.Plugin, bool)
+	getPluginMetadata(plgname string) (plugins_common.IPluginMetadata, bool)
+}
+
+// Interface  for lomipc.ClientTx
+type IClientTx interface {
+	RegisterClient(client string) error
+	DeregisterClient() error
+	RegisterAction(action string) error
+	DeregisterAction(action string) error
+	RecvServerRequest() (*lomipc.ServerRequestData, error)
+	SendServerResponse(res *lomipc.MsgSendServerResponse) error
+	NotifyHeartbeat(action string, tstamp int64) error
+}
+
+type PluginManager struct {
+	clientTx IClientTx
+	plugins  map[string]plugins_common.Plugin /* map : pluginname -> Plugin struct Object(at go/src/plugins/plugins_common)
+	e.g. "linkflapdetection" -->  {linkFlapDetection object} */
+	pluginMetadata map[string]plugins_common.IPluginMetadata /* map : pluginname -> PluginMetadata struct Object(at go/src/plugins/plugins_common) */
+	stopch         chan struct{}                             /* channel used to stop listening on golib for server events */
+	//TODO: Goutham : replace this with global stop channel
+}
+
+// InitPluginManager : Initialize plugin manager & register with server
+func NewPluginManager(clientTx IClientTx) (*PluginManager, error) {
+	if pluginMgr != nil {
+		return pluginMgr, errors.New("client already registered")
+	}
+
+	pluginMgr = &PluginManager{
+		clientTx:       clientTx,
+		plugins:        make(map[string]plugins_common.Plugin),
+		pluginMetadata: make(map[string]plugins_common.IPluginMetadata),
+		stopch:         make(chan struct{}),
+	}
+
+	if err := pluginMgr.clientTx.RegisterClient(ProcID); err != nil {
+		return nil, err
+	}
+
+	return pluginMgr, nil
+}
+
+// getPluginManager : Get plugin manager object
+func (plmgr *PluginManager) getPlugin(plgname string) (plugins_common.Plugin, bool) {
+	plugin, ok := plmgr.plugins[plgname]
+	return plugin, ok
+}
+
+// getPluginMetadata : Get plugin metadata object
+func (plmgr *PluginManager) getPluginMetadata(plgname string) (plugins_common.IPluginMetadata, bool) {
+	pluginmetadata, ok := plmgr.pluginMetadata[plgname]
+	return pluginmetadata, ok
+}
+
+/*
+ * Listens for server requests via golib. This call blocks.
+ *
+ * Input:
+ *  stop - stop channel used to stop the blocking receive server call
+ *
+ * Output:
+ *  none -
+ *
+ * Return:
+ *  error - error message or nil on success
+ */
+func (plmgr *PluginManager) run() error {
+	for {
+		select {
+		case <-plmgr.stopch:
+			lomcommon.LogInfo("Stopping plugin manager run loop")
+			return nil
+		default:
+			serverReq, err := plmgr.clientTx.RecvServerRequest()
+			if err != nil {
+				lomcommon.LogError("Error RecvServerRequest() : %s", err)
+				return err
+			}
+			if serverReq == nil {
+				return lomcommon.LogError("Error RecvServerRequest() : nil")
+			}
+
+			switch serverReq.ReqType {
+			case lomipc.TypeServerRequestAction:
+				actionReq, ok := serverReq.ReqData.(*lomipc.ActionRequestData)
+				if !ok {
```

Comment:
if actionReq, ok....; !ok {
     not ok
} else  if plugin, ok := ....; !ok {
    not ok
} else {
     ok case
}
}

Reply:
modified